### ISSUE TITLE
No Applescript Widgets work - Alpha

### ISSUE THREAD

**Issue by OrbitalTech** [folivoraAI/BetterTouchTool#2409]
#### Description of bug/feature request/question:
Please only post one bug or feature request per issue.

My touchbar's Applescript widgets do not show. When fiddling and changing between Master, and the two latest alphas, it seems none of my applescript widgets display anymore. When running the code inside BTT like so:
<img width="824" alt="screen shot 2018-04-07 at 2 13 13 pm" src="https://user-images.githubusercontent.com/23006558/38450193-da390d02-3a6d-11e8-9a77-2242b4bef1d5.png">
All is fine, but upon clicking save, this happens.
<img width="1085" alt="touch bar shot 2018-04-06 at 7 23 55 pm" src="https://user-images.githubusercontent.com/23006558/38450205-05e1589c-3a6e-11e8-9d4b-5a81ff20e0ba.png">
^Image of my Custom Applescript Clock^ - vImage of touchbar after clicking savev
<img width="1085" alt="touch bar shot 2018-04-07 at 2 13 51 pm" src="https://user-images.githubusercontent.com/23006558/38450211-1ea47b2a-3a6e-11e8-9c8f-387d94405eb4.png">
The correct information is only displayed temporarily, then the widget disappears.

You'll also notice that I no longer have a pause/play button, because I created a custom one using applescript.

#### Affected input device (e.g. MacBook Trackpad, Magic Mouse/Trackpad, Touch Bar, etc.):
Touchbar

#### Device information:

Type of Mac: Macbook Pro 2017 512Gb with Touchbar
macOS version: 10.13.3
BetterTouchTool version: 2.428

**Comment by fifafu** [comment]
That's weird I can't seem to reproduce this using my Apple Scripts.
You are not updating them via the external triggers, they just hide after a few moments?

If possible go to the about tab in BTT and click the "export debug information" button. This will create a folder called BetterTouchToolDebug on your Desktop. Would be great if you could send that to andreas@folivora.ai then I'll look into it.

**Comment by fifafu** [comment]
Weird, for me only the date time widget doesn't work. 
Your spotify script worked fine after I added the status.txt file.
The Play/Pause didn't work for me, but I think it was only missing the default icon? After adding that it seems to work

Now my Touch Bar looks like this with the modified date time widget and play pause widget: 

![image](https://user-images.githubusercontent.com/8820811/38453118-b830195c-3a50-11e8-9b50-28de8f062049.png)

Btw. I'd really recommend to use a little python script using the shell script widget for stuff like the date widget. It is much more performant and easier than Apple Script:
```
import time
import datetime
today = datetime.date.today()
print time.strftime('%a, %I:%M:%S %p') + ' - Day ' + str(today.weekday()+1)
````
![image](https://user-images.githubusercontent.com/8820811/38453061-af15ec30-3a4f-11e8-81a2-ea83c9a34a11.png)
![image](https://user-images.githubusercontent.com/8820811/38453115-b2c77dfc-3a50-11e8-96e7-c3cee7839cf1.png)

**Comment by fifafu** [comment]
However I noticed some weird stuff when adding the pause icon, I'm currently looking into that.

**Comment by fifafu** [comment]
Btw.: don't invest too much time today working with the alpha implementation. Tomorrow I'll probably improve big parts of it which will probably bring some breaking changes.

**Comment by OrbitalTech** [comment]
Awesome. Thanks for that, I'll look into using the python alternative. The day number is actually the current day number for my school's timetable, hence why it reads from a file. (Written by a Java App I wrote for class).

As for the icons, the Pause Play icons are still there for me, but I will make new ones given that my existing ones are so bad anyway.

As for Spotify, not working there, but I'll make sure to try creating it all anew and seeing what happens.

I'm guessing what has happened here is that when using Alpha 2.426 I set the icon to be the shopping basket in your example, which has more than likely caused issues with the regex comparison.

Thanks for your help! I'm very excited for these new releases and the flexibility they bring!
